lms/djangoapps/shoppingcart/models.py

```diff
@@ -1332,6 +1332,12 @@ def add_to_order(cls, order, course_id, mode_slug=CourseMode.DEFAULT_MODE_SLUG,
         # First a bunch of sanity checks:
         # actually fetch the course to make sure it exists, use this to
         # throw errors if it doesn't.
+        course_modes = CourseMode.modes_for_course_dict(course_id)
+        # Check if the NO_ID_PROFESSIONAL mode is defined for the given course
+        # then use this mode for registration instead of default 'honor' mode to
+        # use the correct currency and amount for registration
+        if CourseMode.NO_ID_PROFESSIONAL_MODES[0] in course_modes:
+            mode_slug = CourseMode.NO_ID_PROFESSIONAL_MODES[0]
```

Comment:
This should be handled at a higher level in verify_student `create_order`.  That view should pass in the correct value for `mode_slug` when the course has a no-ID professional mode.